salt/modules/yumpkg.py

```diff
@@ -923,9 +922,11 @@ def list_downloaded():
     CACHE_DIR = os.path.join('/var/cache/', _yum())
 
     ret = {}
-    for package_path in glob.glob(os.path.join(CACHE_DIR, '*/*/*/packages/*.rpm')):
-        pkg_info = __salt__['lowpkg.bin_pkg_info'](package_path)
-        ret.setdefault(pkg_info['name'], {})[pkg_info['version']] = package_path
+    for root, dirnames, filenames in os.walk(CACHE_DIR):
+        for filename in fnmatch.filter(filenames, '*.rpm'):
+            package_path = os.path.join(root, filename)
+            pkg_info = __salt__['lowpkg.bin_pkg_info'](package_path)
+            ret.setdefault(pkg_info['name'], {})[pkg_info['version']] = package_path
```

Comment:
I am not really fan of mapping physical path of the file to a version of the package. It just doesn't sound/look right, because in both sides you _assuming_ the version to path, instead of getting them. You also probably want another stuff to be added like size or timestamp when it had happened, since there are no other history available:

```yaml
your-great-package:
  24.0:
     path: /var/cache/zypp/packages/SLE-12-SP2-x86_64-Pool/x86_64/emacs-rules-the-world-24.0.rpm
     size: 14 Mb
     timestamp_ux: 1492602141
```

etc.